pkg/ui/terminal.go

```diff
@@ -70,42 +70,51 @@ func NewTerminalUI(doc *Document, journal journal.Recorder, useTTYForInput bool)
 		useTTYForInput:   useTTYForInput, // Store this flag
 	}
 
-	// Initialize input handler based on mode
-	if useTTYForInput {
-		// Initialize TTY input
-		tty, err := os.OpenFile("/dev/tty", os.O_RDWR, 0)
-		if err != nil {
-			return nil, fmt.Errorf("opening tty for input: %w", err)
-		}
-		u.ttyFile = tty // Store file handle for closing
-		u.ttyReader = bufio.NewReader(tty)
-	} else {
-		// Initialize readline input
-		historyPath := filepath.Join(os.TempDir(), "kubectl-ai-history")
-		rl, err := readline.NewEx(&readline.Config{
-			Prompt:      ">>> ", // Default prompt for main input
-			Stdin:       os.Stdin,
-			Stdout:      os.Stdout,
-			Stderr:      os.Stderr,
-			HistoryFile: historyPath,
-			// History enabled by default
-		})
-		if err != nil {
-			// Log warning or fallback if readline init fails?
-			klog.Warningf("Failed to initialize readline, input might be limited: %v", err)
-			// Proceed without readline for now, or return error?
-			// Returning error to make it explicit
-			return nil, fmt.Errorf("creating readline instance: %w", err)
-		}
-		u.rlInstance = rl // Store readline instance
-	}
-
 	subscription := doc.AddSubscription(u)
 	u.subscription = subscription
 
 	return u, nil
 }
 
+func (u *TerminalUI) ttyReader() (*bufio.Reader, error) {
+	if u.ttyReaderInstance != nil {
+		return u.ttyReaderInstance, nil
+	}
+	// Initialize TTY input
+	tty, err := os.OpenFile("/dev/tty", os.O_RDWR, 0)
+	if err != nil {
+		return nil, fmt.Errorf("opening tty for input: %w", err)
+	}
+	u.ttyFile = tty // Store file handle for closing
+	u.ttyReaderInstance = bufio.NewReader(tty)
+	return u.ttyReaderInstance, nil
+}
+
+func (u *TerminalUI) readlineInstance() (*readline.Instance, error) {
+	if u.rlInstance != nil {
+		return u.rlInstance, nil
+	}
+	// Initialize readline input
+	historyPath := filepath.Join(os.TempDir(), "kubectl-ai-history")
+	rl, err := readline.NewEx(&readline.Config{
+		Prompt:      ">>> ", // Default prompt for main input
+		Stdin:       os.Stdin,
+		Stdout:      os.Stdout,
+		Stderr:      os.Stderr,
+		HistoryFile: historyPath,
+		// History enabled by default
+	})
+	if err != nil {
+		// Log warning or fallback if readline init fails?
+		klog.Warningf("Failed to initialize readline, input might be limited: %v", err)
+		// Proceed without readline for now, or return error?
+		// Returning error to make it explicit
```

Comment:
Yes, I think returning an error is correct at least until we figure out when this might fail.  If/when we find that, we might want an option to turn off readline.